src/System.Private.CoreLib/shared/System/String.cs

```diff
@@ -655,6 +651,30 @@ internal static unsafe int wcslen(char* ptr)
             return count;
         }
 
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe int strlen(byte* ptr)
+        {
+            if (ptr == null)
+            {
+                return 0;
+            }
+
+            // IndexOf processes memory in aligned chunks, and thus it won't crash even if it accesses memory beyond the null terminator.
+            int length = SpanHelpers.IndexOf(ref *ptr, (byte)'\0', int.MaxValue);
+            if (length < 0)
+            {
+                ThrowMustBeNullTerminatedString();
+            }
+
+            return length;
+        }
+
+        private static void ThrowMustBeNullTerminatedString()
```

Comment:
Change wcslen to call this method too?

Reply:
Yes, was going to do that in the follow up PR